x-pack/elastic-agent/pkg/agent/application/info/agent_id.go

```diff
@@ -21,13 +21,13 @@ import (
 
 // defaultAgentConfigFile is a name of file used to store agent information
 const defaultAgentConfigFile = "fleet.yml"
-const agentInfoKey = "agent_info"
+const agentInfoKey = "agent"
```

Comment:
note: this is breaking between 7.8 and 7.9 so maybe we should put a note that agent might change ID in between versions when updating from 7.8

Reply:
should we just handle it? support reading it with either `agent_info` or `agent`?

Reply:
Lets document it, breaking changes are ok from 7.8->7.9

Reply:
Created #19488 that handles it transparently, or we can just document. I prefer to just handle it, but okay with breaking.